x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/allocation/TrainedModelAllocationClusterService.java

```diff
@@ -249,6 +253,10 @@ static ClusterState updateModelRoutingTable(ClusterState currentState, UpdateTra
         if (existingAllocation == null) {
             throw new ResourceNotFoundException("allocation for model with id [" + modelId + "] not found");
         }
+        // If we are stopping, don't update anything
+        if (existingAllocation.getAllocationState().equals(AllocationState.STOPPING)) {
+            return currentState;
```

Comment:
A debug log statement might be helpful here.

Reply:
👍